cirq/google/api/v2/results.py

```diff
@@ -138,10 +139,9 @@ def results_to_proto(
                     qmr.results = pack_bits(m_data[:, i])
     return out
 
-
 def results_from_proto(
         msg: result_pb2.Result,
-        measurements: List[MeasureInfo],
+        measurements: List[MeasureInfo] = None,
```

Comment:
I think instead of making this optional and relying on ordering, we should use the `api_v2.find_measurements` function to find measurements in the program before sending to the quantum engine, and store this in the `EngineJob`. Then when getting results the list of measurements can be passed in to `results_from_proto`.

Reply:
Shouldn't all the info about the measurements be accessible from the results?

Reply:
What is your concern with relying on the order in the result? If we want to support re-ordering, could it be a feature/utility rather than a requirement to parse a result?

Reply:
We want to be sure that the order in which results appear in `TrialResult` match the original measurement operation. The `measurements` arg on `results_from_proto` is here so we can check the qubit order in the results against the original measure ops, and it's as easy to reorder as to check so that's what it does. If we want to specify that the engine will not reorder (and then assume but not check this), then I'd suggest we remove this argument entirely rather than making it optional.

I think it would be good to include the qubit order in `TrialResult` itself to make the results more self-describing. That would alleviate some of my concern because we won't have so many assumptions about parallel ordering between various objects. I know `TrialResult` is getting an overhaul, so maybe we can include the qubit order as part of that change.

Reply:
+1 to including qubit order details in the result.

I'd prefer to remove the `measurements` arg, but that would be a breaking change for existing external dependencies (I found one). My plan was to make this optional to continue support, then, if it was acceptable, remove all references, then clean up by removing the param. LMK if it's fine to just make the breaking change, otherwise I can add comments to that effect.

Reply:
I agree with @dabacon, making this optional for now is fine. We can remove in a future PR.